mypy/semanal.py

```diff
@@ -5125,6 +5226,93 @@ def visit_match_stmt(self, s: MatchStmt) -> None:
                 guard.accept(self)
             self.visit_block(s.bodies[i])
 
+    def visit_type_alias_stmt(self, s: TypeAliasStmt) -> None:
+        self.statement = s
+        type_params: TypeVarLikeList = []
+        all_type_params_names = []
+        for p in s.type_args:
+            upper_bound = p.upper_bound or self.object_type()
+            fullname = self.qualified_name(p.name)
+            type_params.append(
+                (
+                    p.name,
+                    TypeVarExpr(
+                        p.name,
+                        fullname,
+                        [],
+                        upper_bound,
+                        default=AnyType(TypeOfAny.from_omitted_generics),
+                    ),
+                )
+            )
+            all_type_params_names.append(p.name)
+
+        self.push_type_args(s.type_args, s)
```

Comment:
Maybe defer if `push_type_args` returned `False` e.g. when a forward reference is used as upper_bound? This would fix [Issue 4](https://github.com/python/mypy/pull/17233#issuecomment-2106031542).

Reply:
Yes, this works. Fixed the issue.